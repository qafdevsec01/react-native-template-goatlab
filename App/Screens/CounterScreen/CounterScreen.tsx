import React from 'react'
import { Col, Grid, Row } from 'react-native-easy-grid'
import { Button, Text } from 'react-native-elements'
import { useDispatch, useSelector } from 'react-redux'
import Container from '../../Components/Container'
import Header from '../../Components/Header'
import { counterSelector } from '../../Redux/CounterRedux'
import CounterActions from '../../Redux/CounterRedux'
import styles from './CounterScreenStyles'

export default props => {
  const counter = useSelector(counterSelector)
  const dispatch = useDispatch()
  const actions = {
    increase: () => dispatch(CounterActions.increase()),
    decrease: () => dispatch(CounterActions.decrease())
  }
  return (
    <>
      <Header />
      <Container>
        <Grid style={styles.grid}>
          <Row style={styles.row}>
            <Col style={styles.button}>
              <Button title="-" onPress={() => actions.decrease()} />
            </Col>
            <Col style={styles.center}>
              <Text style={styles.text}>Counter: {counter}</Text>
            </Col>
            <Col style={styles.button}>
              <Button title="+" onPress={() => actions.increase()} />
            </Col>
          </Row>
        </Grid>

        <Button title="Go back" onPress={() => props.navigation.goBack()} />
      </Container>
    </>
  )
}
