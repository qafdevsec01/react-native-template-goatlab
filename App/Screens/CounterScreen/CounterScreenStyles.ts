import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  button: {
    width: 40,
    flex: 1,
    flexGrow: 1,
    justifyContent: 'center'
  },
  buttonText: {
    color: 'white'
  },
  center: {
    flex: 4,
    alignItems: 'center',
    alignContent: 'center',
    justifyContent: 'center'
  },
  row: {
    height: 100
  },
  text: {
    fontSize: 32,
    fontWeight: 'bold'
  },
  grid: {
    justifyContent: 'center'
  }
})
