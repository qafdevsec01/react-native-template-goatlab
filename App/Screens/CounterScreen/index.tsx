import React from 'react'
import { Text } from 'react-native'

export default props => {
  return <Text>Hola</Text>
}
