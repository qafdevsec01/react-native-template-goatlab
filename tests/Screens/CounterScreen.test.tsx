import { light, mapping } from '@eva-design/eva'
import React from 'react'
import { ApplicationProvider } from 'react-native-ui-kitten'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import CounterScreen from 'appSrc/Screens/CounterScreen'
import createStore from '../Mocks/createStore'

const createTestProps = (props: {}) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

const store = createStore()

test('CounterScreen renders correctly', () => {
  let props: any
  props = createTestProps({})
  const tree = renderer
    .create(
      <Provider store={store}>
        <ApplicationProvider mapping={mapping} theme={light}>
          <CounterScreen {...props} />
        </ApplicationProvider>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
