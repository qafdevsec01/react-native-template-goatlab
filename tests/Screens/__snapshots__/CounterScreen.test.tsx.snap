// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterScreen renders correctly 1`] = `
Array [
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
          }
        }
      >
        <View
          icon="menu"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#fff",
          }
        }
        text="GOATLAB'S BOILERPLATE"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        GOATLAB'S BOILERPLATE
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          icon="home"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          underlayColor="white"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0,
                "flexDirection": "column",
                "width": 40,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    -
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Counter: 
              0
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 0,
                "flexDirection": "column",
                "width": 40,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingBottom": 1,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    +
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Go back
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>,
]
`;
